exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -110,6 +118,50 @@ public Viewable submitQuery(@FormParam("query") String query,
     }
   }
 
+  private Map<String, Object> readOptionsFromForm(Form form) {
+    Map<String, Object> options = new HashMap<>();
+    SessionOptionManager sessionOptionManager = webUserConnection.getSession().getOptions();
+    for (Map.Entry<String, List<String>> pair : form.asMap().entrySet()) {
+      String name = pair.getKey();
+      OptionDefinition definition = sessionOptionManager.getOptionDefinition(name);
+      if (definition != null) {
+        String valueStr = pair.getValue().get(0);
```

Comment:
Can't `valueStr` be null?

Reply:
I can't really think of a circumstance where the form parser insert a key with zero values, it doesn't make any sense to me that it would do that, since it only inserts keys for values it finds in the Form and thus there should be at least one value for each key.